soql-server-pg/src/main/scala/com/socrata/pg/server/QueryServer.scala

```diff
@@ -161,16 +161,13 @@ class QueryServer(val dsInfo: DSInfo, val caseSensitivity: CaseSensitivity, val
     def exists(req: HttpRequest): HttpResponse = {
       val servReq = req.servletRequest
 
-      val dsOp = Option(servReq.getParameter("ds"))
+      val ds = Option(servReq.getParameter("ds")).getOrElse(throw new IllegalArgumentException("Existence Checking requires the dataset parameter"))
```

Comment:
That'll 500.  If you want to make it required, make it not a query parameter at all - `/exists/{ds}` instead of `/exists?ds=...`